internal/util/kubernetes/kubernetes.go

```diff
@@ -86,10 +90,32 @@ func GetSecret(ctx context.Context, client k8sClient.Client, namespace, secretNa
 	return secret, nil
 }
 
+func DeleteKubernetesResource(ctx context.Context, client k8sClient.Client, item k8sClient.Object) error {
+	numaLogger := logger.FromContext(ctx)
+	if err := client.Delete(ctx, item); err != nil {
+		if apierrors.IsNotFound(err) {
+			numaLogger.Info("Object not found", item)
+			return nil
+		}
+		return fmt.Errorf("error deleting resource %s/%s: %v", item.GetNamespace(), item.GetName(), err)
+	}
+	return nil
+}
+
 func IsValidKubernetesManifestFile(fileName string) bool {
 	fileExt := strings.Split(fileName, ".")
 	if _, ok := validManifestExtensions[fileExt[len(fileExt)-1]]; ok {
 		return true
 	}
 	return false
 }
+
+// DeleteManagedObjectsGitSync deletes Kubernetes resources from a map sequentially, returning an error if any deletion fails.
+func DeleteManagedObjectsGitSync(ctx context.Context, client k8sClient.Client, objs map[kube.ResourceKey]*unstructured.Unstructured) error {
```

Comment:
nit: let's change the name of this function because it actually has nothing to do with `GitSync` per se. you're just deleting some resources